r/.lintr

```diff
@@ -1,23 +1,8 @@
-license:  '#  Licensed to the Apache Software Foundation (ASF) under one
-  #  or more contributor license agreements.  See the NOTICE file
-  #  distributed with this work for additional information
-  #  regarding copyright ownership.  The ASF licenses this file
-  #  to you under the Apache License, Version 2.0 (the
-  #  "License"); you may not use this file except in compliance
-  #  with the License.  You may obtain a copy of the License at
-  #
-  #   http://www.apache.org/licenses/LICENSE-2.0
-  #
-  #  Unless required by applicable law or agreed to in writing,
-  #  software distributed under the License is distributed on an
-  #  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-  #  KIND, either express or implied.  See the License for the
-  #  specific language governing permissions and limitations
-  #  under the License.'
 linters: linters_with_defaults(
   indentation_linter = NULL,
   line_length_linter = line_length_linter(120),
   object_name_linter = NULL,
+  return_linter = NULL,
```

Comment:
I could have updated our code, but IMO the places where we have explicit returns actually enhance readability, so just disabled this linter instead.

Reply:
Do you have an example of where this complained / where it's more explicit? I'm fine to turn it off, but would be good to have a note here for history

Reply:
Yup, here are a couple:

https://github.com/apache/arrow/blob/ec3d2839d102cc44d368c154cfa756eb946909ee/r/R/dplyr-summarize.R#L302-L312

and 

https://github.com/apache/arrow/blob/ec3d2839d102cc44d368c154cfa756eb946909ee/r/R/install-arrow.R#L270

Reply:
_nods_ got it. It's funny cause both of those IIUC do follow the "implicit return, unless you're returning early" rule. I guess there might be some room for interpretation around if the last thing is an `if`/`else` then that's the same thing as the last thing.

Agreed that these we should keep as they are. Are there a small enough number of these kinds of false positives that we could use the `except` or `except_regex` option to except them? https://lintr.r-lib.org/reference/return_linter.html#arg-except-except-regex I'm fine if we don't but that's one way to get some of the niceness of the linter still

Reply:
I like the idea, but it doesn't look like it, as the docs say that those options are only applicable when the return style is set to `explicit`.